src/subscription_manager/cli.py

```diff
@@ -196,11 +194,11 @@ def main(self) -> Optional[int]:
             sys.exit(return_code)
 
         try:
-            return cmd.main()
+            cmd.main()
+            return 0
         except InvalidCLIOptionError as error:
-            # FIXME Re-raise the exception instead. Returning None
-            #  (a non-integer) results in return code 0, which is not correct.
```

Comment:
or, IMHO even better:
- simply do not catch `InvalidCLIOptionError` here at all
- in the `main()` in `src/subscription_manager/scripts/subscription_manager.py`, catch `InvalidCLIOptionError` (just like done with `KeyboardInterrupt`) using `system_exit(1, exc)` to print it and exit